src/NuGet.Clients/NuGet.Credentials/CredentialService.cs

```diff
@@ -34,7 +34,9 @@ public class CredentialService : ICredentialService
 
         private Action<string> ErrorDelegate { get; }
 
-        public bool HandlesDefaultCredentials { get; }
+        public async Task<bool> HandlesDefaultCredentialsAsync() {
+            return (await _providers).Any(provider => provider is DefaultCredentialsCredentialProvider);
```

Comment:
where ever possible use `Microsoft.VisualStudio.Threading.AsyncLazy` instead of `NuGet.Common.AsyncLazy` which isn't a real async lazy.